### PR TITLE
Drop blockenspiel

### LINKED ISSUE #3231: Assess and maybe replace blockenspiel gem
## Description

Our dependencies need to be kept up to date. Since that is work, we only want to keep dependencies that save lots of work in other ways. A discussion in https://github.com/openfoodfoundation/openfoodnetwork/pull/3211 opened the question if we need the gem blockenspiel. We need to assess its use and decide if we want to keep it.

## Severity

Low tech debt.

## Possible Fix

Two possible options:

- Replace the current logic using blockenspiel and remove the gem.
- Write a guide how to use blockenspiel efficiently in our code base.

### PR DESCRIPTION
**Pull request by cillian**
#### What? Why?

Closes #3231

:wave: 
Here is a proposal that removes the Blockenspiel `organise` and `summary_row` DSL because most other reports seem to use the `#rules` and `#columns` methods instead. 

It also adds the option to define report columns and summary rows with Symbols instead of Procs so `Proc { ... }` doesn't need to be repeated for each column. Other reports that use Procs should still work.

#### What should we test?

The four report types in the Admin > Reports > Bulk Co-Op section e.g.

Bulk Co-op - Totals by Supplier
Bulk Co-op - Allocation
Bulk Co-op - Packing sheets
Bulk Co-op - Customer payments

#### Release notes

Drop Blockenspiel gem and DSL and use :rules and :columns methods for consistency with other reports.

Changelog Category: Changed

### CODE REVIEW

File: lib/open_food_network/bulk_coop_report.rb [hunk 1]
```diff
@@ -134,5 +140,127 @@ def order_permissions
     def report_line_items
       @report_line_items ||= Reports::LineItems.new(order_permissions, @params)
     end
+
+    def customer_payments_total_cost(line_items)
```
Comment (luisramos0) [inline lib/open_food_network/bulk_coop_report.rb:144]: All these methods below could be extracted to a service. Some of them are repeated in other reports I am sure.

File: lib/open_food_network/bulk_coop_report.rb [hunk 2]
```diff
@@ -153,7 +153,7 @@ def customer_payments_amount_paid(line_items)
       line_items.map(&:order).uniq.sum(&:payment_total)
     end
 
-    def empty_cell(line_items)
+    def empty_cell(*)
```
Comment (luisramos0) [inline lib/open_food_network/bulk_coop_report.rb]: I didnt know this existed in Ruby :see_no_evil: 
I think I would prefer to see an indication of what could be sent and not used here. So _line_items would be better than * imho.

**Comment by cillian** [comment]
Ah good, yes I was about to mention that spec because it was passing locally....thanks

**Comment by sauloperez** [comment]
I resolved the merge conflict from the UI so we can move to test ready.

**Comment by sigmundpetersen** [comment]
> Don't worry about broken spec spec/lib/open_food_network/products_and_inventory_report_spec.rb:110
> It's a flaky spec that I have removed in a recent PR yet to be merged.

This failure seems quite consistent through this PR. 
@luisramos0 do you remember in which PR this was removed/fixed?

**Comment by luisramos0** [comment]
yes, here:
https://github.com/openfoodfoundation/openfoodnetwork/pull/5446/commits/664d01aa882ed60f1b59d7c103c97227cbe2bbff

**Comment by filipefurtad0** [comment]
Hey @sauloperez and @luisramos0 ,
This PR has conflicts with master - perhaps best to solve these before proceeding with the tests?

The codeclimate issue is a flaky-spec, and safe to ignore, right?
KNAPSACK_TEST_DIR=spec KNAPSACK_TEST_FILE_PATTERN="{spec,engines/**/spec}/**{,/*/**}/*_spec.rb" bundle exec rake 'knapsack:rspec[--tag ~performance --format progress -p]'

Thanks!

**Comment by cillian** [comment]
Hi @filipefurtad0 I pushed a fix to resolve those conflicts now. Let me know if you want me to do a rebase or anything.

**Comment by filipefurtad0** [comment]
Hi @cillian ,
Thanks for solving the branch conflicts!

I checked the rendering of the reports:

Bulk Co-op - Totals by Supplier
Bulk Co-op - Allocation
Bulk Co-op - Packing sheets
Bulk Co-op - Customer payments

Before and after your PR, across instances (as superadmin): they are identical, as well as the downloadable files. New purchases appear in the reports as expected.

All good, ready to go.

**Comment by luisramos0** [comment]
great :clap: :clap: :clap:

**Comment by sauloperez** [comment]
Thanks for the patience and this pretty useful contribution @cillian !

**Pull request merged**